independent-projects/arc/runtime/src/main/java/io/quarkus/arc/Unremovable.java

```diff
@@ -0,0 +1,15 @@
+package io.quarkus.arc;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Indicates that the class marked with this annotation should never be removed by Arc as a bean
+ * even if it has no known injection points
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
```

Comment:
We should add `ElementType.METHOD` and `ElementType.FIELD`  and make it possible to annotate a producer method/field.

Reply:
Hadn't thought of that :). So in this case we would need to do some extra work to figure out which class is unremovable, correct?

Reply:
Should we do the extra processing work in this PR do you think or leave it as an enhancement for later?

Reply:
If we want it part of this PR, I can most likely spare some time to do it later on today.

Reply:
Hm, it shouldn't require any extra work. Just discover all `@Unremovable` annotations and match the `AnnotationInstance.target()` with `BeanInfo.getTarget()` .

Reply:
Ah yes! I forgot that `AnnotationInstance.target()` would work for all types. I'll fix the PR soon then :)

Reply:
Fixed